### PR TITLE
Add Scala 3

### PR DESCRIPTION
**Pull request by ybasket**
Replaces `circe-generic-extras` with circe-generic by renaming fields to snake_case. Adds a custom Arbitrary instance to work around a missing `scalacheck-shapeless` dependency on Scala 3.

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -4,9 +4,10 @@ object Dependencies {
   object Versions {
     val scala212 = "2.12.14"
     val scala213 = "2.13.6"
+    val scala3 = "3.0.1"
 
-    val trace4cats = "0.12.0-RC2"
-    val trace4catsExporterHttp = "0.12.0-RC2"
+    val trace4cats = "0.12.0-RC2+17-d73c7ff3"
+    val trace4catsExporterHttp = "0.12.0-RC2+4-5c079741"
```
Comment (ybasket) [inline project/Dependencies.scala:10]: This one isn't released yet, CI will hence fail.

File: modules/stackdriver-http-exporter/src/main/scala/io/janstenpickle/trace4cats/stackdriver/oauth/GoogleAccountParser.scala [hunk 2]
```diff
@@ -11,20 +11,20 @@ import java.util.regex.Pattern
...
 
 object GoogleAccountParser {
   case class JsonGoogleServiceAccount(
     `type`: String,
-    projectId: String,
-    privateKeyId: String,
-    privateKey: String,
-    clientEmail: String,
-    authUri: String
+    project_id: String,
```
Comment (ybasket) [inline modules/stackdriver-http-exporter/src/main/scala/io/janstenpickle/trace4cats/stackdriver/oauth/GoogleAccountParser.scala:19]: Some of them seem unused, so they could be removed. Mirroring the snake_case of the input is the easiest way to work around the missing circe-generic-extras, but it's also possible to write a `Decoder` by hand (the format should basically never change) or try to pre-process the JSON before decoding. Let me know if you prefer a different approach.

File: project/Dependencies.scala (hunk 1 above)
Comment (catostrophe) [reply project/Dependencies.scala:10]:
```suggestion
    val trace4catsExporterHttp = "0.12.0-RC2+4-69bfcea6"
```

**Comment by catostrophe** [comment]
Not sure if changing the model fields is ok. Seems true but let's ask @janstenpickle for sure.

**Review by catostrophe — APPROVED** [review]

File: modules/stackdriver-http-exporter/src/main/scala/io/janstenpickle/trace4cats/stackdriver/oauth/GoogleAccountParser.scala (hunk 2 above)
Comment (janstenpickle) [reply modules/stackdriver-http-exporter/src/main/scala/io/janstenpickle/trace4cats/stackdriver/oauth/GoogleAccountParser.scala:19]: I ported these from another older library, so I don't think the extra fields are needed anymore.

I think this approach for snake case should be fine.

**Pull request merged**